testframework/src/main/java/net/neoforged/testframework/junit/EphemeralTestServerProvider.java

```diff
@@ -0,0 +1,318 @@
+import java.util.function.BooleanSupplier;
+import net.minecraft.SystemReport;
+import net.minecraft.Util;
+import net.minecraft.commands.Commands;
+import net.minecraft.core.MappedRegistry;
+import net.minecraft.core.Registry;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.Services;
+import net.minecraft.server.WorldLoader;
+import net.minecraft.server.WorldStem;
+import net.minecraft.server.level.progress.LoggerChunkProgressListener;
+import net.minecraft.server.packs.repository.PackRepository;
+import net.minecraft.server.packs.repository.ServerPacksSource;
+import net.minecraft.server.players.PlayerList;
+import net.minecraft.util.datafix.DataFixers;
+import net.minecraft.util.debugchart.LocalSampleLogger;
+import net.minecraft.util.debugchart.SampleLogger;
+import net.minecraft.world.Difficulty;
+import net.minecraft.world.flag.FeatureFlags;
+import net.minecraft.world.level.DataPackConfig;
+import net.minecraft.world.level.GameRules;
+import net.minecraft.world.level.GameType;
+import net.minecraft.world.level.LevelSettings;
+import net.minecraft.world.level.WorldDataConfiguration;
+import net.minecraft.world.level.dimension.LevelStem;
+import net.minecraft.world.level.levelgen.WorldDimensions;
+import net.minecraft.world.level.levelgen.WorldOptions;
+import net.minecraft.world.level.levelgen.presets.WorldPresets;
+import net.minecraft.world.level.storage.LevelStorageSource;
+import net.minecraft.world.level.storage.PrimaryLevelData;
+import net.neoforged.neoforge.server.ServerLifecycleHooks;
+import org.apache.logging.log4j.LogManager;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.slf4j.Logger;
+
+// @formatter:off
+/**
+ * A {@link ParameterResolver} that provides a {@link MinecraftServer} parameter.
+ * <p>
+ * The server is ephemeral, meaning that it doesn't store any data, and only has a void overworld available.
+ * <p>
+ * You should <strong>NOT</strong> not interact with the world of that server as it purely exists to load datapack data.
+ * If you need an actual world, you should use a {@linkplain net.minecraft.gametest.framework.GameTest GameTest} instead.
+ *
+ * <p>
+ * Example usage:
+ * {@snippet :
+ * @Test
+ * @ExtendWith(EphemeralTestServerProvider.class)
+ * void someJUnitTest(MinecraftServer server) {
+ *     assert server.registryAccess().registryOrThrow(Registries.ITEM).getTag(ItemTags.ANVIL).isPresent();
+ * }
+ * }
+ * You can also annotate a class with {@link ExtendWith} to provide a server to all tests in that class.
+ *
+ * <p>
+ * The server instance is lazy (only created if a test needs it) and <strong>singleton</strong>.
```

Comment:
Just to clarify, the singleton server exists for the entire JUnit session, correct?